pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/PrestoDateTrunc.java

```diff
@@ -0,0 +1,204 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.core.operator.transform.function;
+
+import org.apache.pinot.core.common.DataSource;
+import org.apache.pinot.core.operator.blocks.ProjectionBlock;
+import org.apache.pinot.core.operator.transform.TransformResultMetadata;
+import org.apache.pinot.core.plan.DocIdSetPlanNode;
+import org.joda.time.*;
+import org.joda.time.chrono.ISOChronology;
+import org.joda.time.field.DividedDateTimeField;
+import org.joda.time.field.OffsetDateTimeField;
+import org.joda.time.field.ScaledDurationField;
+
+import javax.annotation.Nonnull;
+import java.util.List;
+import java.util.Map;
+
+import static java.util.Locale.ENGLISH;
+import static org.apache.pinot.core.operator.transform.function.TimeZoneKey.MAX_TIME_ZONE_KEY;
+import static org.apache.pinot.core.operator.transform.function.TimeZoneKey.getTimeZoneKeys;
+
+public class PrestoDateTrunc
```

Comment:
why do we need the Presto prefix? can we not use this without Presto

Reply:
This has nothing to do with Presto and the function can be used independantly. 

I just wanted to distinguish it with the existing dateTimeConvert since it is not compatible with that. 

dateTimeConvert truncates the week to Thursday (5pm PT perhaps !) whereas this function truncates to Monday midnight in the specified timezone. 

What name would you suggest ? How about PrestoCompatibleDateTrunc or SqlCompatibleDateTrunc ? Although I imagine there is nothing "Sql standard" about date truncation.